### PR TITLE
Update dog pregnancy constraints

### PR DESCRIPTION
**Pull request by btjl**

### CODE REVIEW

**Comment by btjl** [comment]
Still debugging the error

```
CONSTRAINT dog_pregnancy_check CHECK (((dog_gender <> 'MALE'::public.t_dog_gender) AND (dog_ever_pregnant <> 'YES'::public.t_yes_no_unknown))),
make: *** [test] Error 1
```

File: migrations/V26_update_dog_constraints.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE dogs
```
Comment (danui) [inline migrations/V26_update_dog_constraints.sql]: The file name should have two underscores to delimit version from name. In this case—`V26__...`

Comment (btjl) [reply migrations/V26_update_dog_constraints.sql]: Ahh thanks!

**Comment by danui** [comment]
Let's also make a corresponding pull request in barkbank-pawtal to verify that this constraint works as intended.

1) In barkbank-pawtal, use `make schema-recv` to rsync the changes from barkbank-schema into pawtal.

2) Write tests in `tests/schema/dog-pregnancy-check.test.ts` for the following:

- it should not be able^2 to insert a MALE dog with pregnancy history UNKNOWN^1
- it should not be able to insert a MALE dog with pregnancy history YES
- it should be able^3 to insert a MALE dog with pregnancy history NO
- it should be able to insert a FEMALE dog with pregnancy history UNKNOWN
- it should be able to insert a FEMALE dog with pregnancy history YES
- it should be able to insert a FEMALE dog with pregnancy history NO

_^1 the `insertDog` test helper function accepts overrides that we can use in this case to set `dogGender` and `dogEverPregnant`_

_^2 the `expectError` test helper can be used to assert an Error is thrown._

_^3 the `expectSuccess` test helper can be used to assert no Error is thrown._

**Review by danui — APPROVED** [review]

**Pull request merged**